doc/user-manual/language/coinduction.lagda.rst

```diff
@@ -111,6 +108,79 @@ Finally, we can prove that ``merge`` is a left inverse for ``split``:
     hd-≡ (merge-split-id _)  = refl
     tl-≈ (merge-split-id xs) = merge-split-id (tl xs)
 
+Coinductive Record Constructors
+-------------------------------
+
+It is possible to give an explicit constructor to coinductive record types like ``Stream``:
+
+::
+
+    record Stream' (A : Set) : Set where
+      coinductive
+      constructor cons
+      field
+        hd : A
+        tl : Stream' A
+
+..
+  ::
+
+    open Stream'
+
+However, this constructor cannot be pattern-matched:
+
+::
+
+    -- Get the third element of a stream
+    third : ∀{A} → Stream' A → A
+
+    -- Not allowed:
+    -- third (cons _ (cons _ (cons x _))) = x
+
+Instead, you can use the record fields as projections:
+
+::
+
+    third str = str .tl .tl .hd
+
+The constructor can be used as usual in the right-hand side of definitions:
+
+::
+
+    -- Prepend a list to a stream
+    prepend : ∀{A} → List A → Stream' A → Stream' A
+    prepend [] str = str
+    prepend (a ∷ as) str = cons a (prepend as str)
+
+However, it doesn't count as 'guarding' for the productivity checker:
+
+::
+
+    -- Make a stream with one element repeated forever
+    cycle : ∀{A} → A → Stream' A
+
+    -- Does not termination-check:
+    -- cycle a = cons a (cycle a)
+
+Instead, you can use copattern matching:
+
+::
+
+    cycle a .hd = a
+    cycle a .tl = cycle a
+
+It is also possible to use copattern-matching lambdas:
+
+::
+
+    cycle' : ∀{A} → A → Stream' A
+    cycle' a = λ where
+      .hd → a
+      .tl → cycle' a
+
+
+For more information on these restrictions, see `this pull request<https://github.com/agda/agda/pull/4611>`,
+and `this commit<https://github.com/agda/agda/commit/d771ac257ee9a1f9662364e5db8a50f9994dac49>`.
```

Comment:
Rendering of these links failed (see https://agda--6353.org.readthedocs.build/en/6353/language/coinduction.html#coinductive-record-constructors).

Reply:
Now fixed (see rendering)